hardhat.config.ts

```diff
@@ -187,11 +187,17 @@ export default {
     currency: "USD",
     gasPrice: 21,
   },
+  sourcify: {
+    enabled: true,
+  },
   etherscan: {
     apiKey: {
       sepolia: ETHERSCAN_API_KEY,
       arbitrumOne: ETHERSCAN_API_KEY,
       base: ETHERSCAN_API_KEY,
+      optimisticEthereum: ETHERSCAN_API_KEY,
+      polygon: ETHERSCAN_API_KEY,
```

Comment:
Did this work for you using the same key as other chains?? 
I tried deploying today with Etherscan API Key (on another project) and it didn't work. 
I created a new key specifically for polygonscan.com and then it did.